src/protocol.vala

```diff
@@ -462,7 +462,7 @@ namespace LanguageServer {
         public bool deprecated { get; set; }
         public Gee.List<CompletionItemTag> tags { get; private set; default = new Gee.ArrayList<CompletionItemTag> (); }
         public string? insertText { get; set; }
-        public InsertTextFormat insertTextFormat { get; set; }
+        public InsertTextFormat? insertTextFormat { get; set; default = null; }
```

Comment:
Again, the commit is fine. The only thing is that you need to remove the `default = null` part, as it's unnecessary.

Reply:
It actually is necessary, at least as long as the `CCode` declaration is there. You can see for yourself by running this:

```vala
[CCode (default_value = "INSERT_TEXT_FORMAT_PlainText")]                                                                                                                                                                                      
enum InsertTextFormat {                                                                                                                                                                                                                       
    PlainText = 1,                                                                                                                                                                                                                            
    Snippt = 2,                                                                                                                                                                                                                               
}                                                                                                                                                                                                                                             
                                                                                                                                                                                                                                              
class CompletionItem : Object, Json.Serializable {                                                                                                                                                                                            
    public InsertTextFormat? insertTextFormat { get; set; }                                                                                                                                                                                   
}                                                                                                                                                                                                                                             
                                                                                                                                                                                                                                              
void main() {                                                                                                                                                                                                                                 
    stdout.printf("%s\n", Json.gobject_to_data(new CompletionItem(), null));                                                                                                                                                                  
}
```

As-is, this gives the following output:

```
{
  "insertTextFormat" : 0
}
```

This is a bug because 0 is not a valid value for this enum, and has the potential to cause clients to break.

This can be fixed by either removing the default value defined in `CCode`, or by explicitly setting the default as null. One of the two is needed, though.

This was tested with Vala 0.48.7.